### ISSUE TITLE
Different validation between attribute option API and Admin form

### ISSUE THREAD

**Issue by mrobichaud-absolunet** [magento/magento2#32608]
Hi,

There seems to be a difference in how magento validate saving attribute option value since 2.4. If you try to put an admin value at "0" from the api, you get the following message from the api "The attribute option label is empty. Enter the value and try again.". But if you go in the admin panel and try to add the value "0" you do not get the same validation error and it lets you add the value. 

The validation is made in Magento\Eav\Model\Entity\Attribute\OptionManagement line 63 and 96 where this bit of code was added

```
$label = trim($option->getLabel() ?: '');
if (empty($label)) {
	throw new InputException(__('The attribute option label is empty. Enter the value and try again.'));
}

```
The problem being that "0" in the $option->getLabel() variable will return '' with this code $option->getLabel() ?: ''. 

The code could be modified to something like this to let pass "0"  if this wasn't the desired effect, or the validation should be added to the admin form to have the same result. 
`($option->getLabel() !== null && $option->getLabel() !== false) ? $option->getLabel() : ''`

The api call in question is POST rest/all/V1/products/attributes/{Attribute_Code}/options where the attribute is a select type
```
{
    "option":
    {
        "label" : "0",
        "value":"0",
        "sort_order":1,
        "store_labels":
            [
                {"storeId":0,"label":"0"},
                {"storeId":11,"label":"0"},
                {"storeId":10,"label":"0"}
            ]
    }
}
```

### Preconditions (*)
1. Magento 2.4

### Steps to reproduce (*)
1. In Admin create new product attribute type 'Multiple Select' and set for it option with admin value "0" and save - attribute saves without any warning.
![Screenshot from 2021-03-29 10-46-50](https://user-images.githubusercontent.com/60198413/112803576-1f46a500-907c-11eb-9097-7d57e08ce944.png)
2. Try to put an admin value at "0" from the REST API, you get the following message from the api "The attribute option label is empty. Enter the value and try again."
The api call is POST rest/all/V1/products/attributes/{Attribute_Code}/options where the attribute is a select type
```
{
    "option":
    {
        "label" : "0",
        "value":"0",
        "sort_order":1,
        "store_labels":
            [
                {"storeId":0,"label":"0"},
                {"storeId":11,"label":"0"},
                {"storeId":10,"label":"0"}
            ]
    }
}
```
### Expected result (*)
The behavior in Admin backend and via REST API must be the same 

### Actual result (*)
The behavior in Admin backend and via REST API is different

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Oscar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-41619` were created

**Issue Available**: @engcom-Oscar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by ProkopovVitaliy** [comment]
@magento I am working on this

**Comment by tufahu** [comment]
> @magento I am working on this

Do you mind if i send a pr for this ? I'm working on a project where it cause problem.

**Comment by pmonosolo** [comment]
I have the same issue when we do product import where new attributes are being created. Has anybody solved this?

**Comment by pmonosolo** [comment]
If anybody is looking for a temporary solution Mirasvit Advanced Sort plugin is working well on 2.4.3.

**Comment by bartoszkubicki** [comment]
It seems to be solved in magento 2.4.5

**Comment by engcom-Bravo** [comment]
Hi @mrobichaud-absolunet,

Thanks for your reporting and collaboration.

We have verified the issue in latest 2.4-develop instance and the issue is no more reproducible.Kindly refer the screenshots.

<img width="1036" alt="Screenshot 2024-10-22 at 09 51 42" src="https://github.com/user-attachments/assets/a30d82d7-c0f1-4865-9819-317195a1f4c9">


The behavior in Admin backend and via REST API is same.

Hence we are closing this issue.

Thanks.
